src/main/cpp/auton/primitives/DrivePath.cpp

```diff
@@ -28,63 +28,106 @@ DrivePath::DrivePath() : m_chassis(SwerveChassisFactory::GetSwerveChassisFactory
                          m_currentChassisPosition(units::meter_t(0), units::meter_t(0), units::radian_t(0))
 {
 }
-
 void DrivePath::Init(PrimitiveParams *params)
 {
+  /* Path Names to load should be definded as followed in xml file Case Sensitive.
+      Bounce1,Bounce2,Bounce3,Bounce4
+      Barrel1
+      GS  
+      Slalom1
 
-  // Read path into trajectory for deploy directory.  JSON File ex. Bounce1.wpilid.json
-  wpi::SmallString<64> deployDir;
-  frc::filesystem::GetDeployDirectory(deployDir);
-  wpi::sys::path::append(deployDir, "paths");
-  wpi::sys::path::append(deployDir, params->GetPathName());
-
-  m_timer->Reset();
- 
-
-  // set current position to initial position which should be first node in path.
-  m_currentChassisPosition = m_trajectory.InitialPose();
-
-  // assume start angel of zero at start of path
-  frc::Rotation2d StartAngle;
-  StartAngle.Degrees() = units::degree_t(0);
-  m_chassis.get()->ResetPosition(m_currentChassisPosition, StartAngle);
-  
-  
-  m_PosChgTimer.Start(); // start scan timer to detect motion
-}
+      Note: GS is for galatic search and will select one of four possible search paths
+      GS_A_Blue
+      GS_A_Red
+      GS_B_Blue
+      GS_B_Red
+*/
+  if (params->GetID() == DRIVE_PATH)
+  {
+    sPath2Load = params->GetPathName(); //Get Path Name from Primitives if Path Name = "GS" then
+                                        // determine which galatic search path to run from network table
+
+    if (sPath2Load == "GS") // If primitives passes "GS"  then get path based on vision results
+    {
+      //routine to Fetch Galactic seach path based on vision results from network table.
```

Comment:
I would have done this differently.   I would have had the auton selector decide which XML file to run using the vision detection.   The XML file would then have the path primitive called out with the correct json file.  This file then only deals with getting a json file from the primitive parameters and running it.   This allows the detection/choice to be running before we enable and this primitive is more generic.   The added benefit is, if we decide that the path following isn't close enough, we can break the path and use vision to do the last little bit before the ball and then switch back to the next path segment.